tests/basic/sharedv4_svc_test.go

```diff
@@ -749,6 +695,157 @@ var _ = Describe("{Shared4SvcUnexportExport}", func() {
+			var replicaNodeIDs map[string]bool
+			var attachedNode, clientNode *node.Node
+			var failover bool
+
+			//set HA level to 2 to predict which node we fail over to (needed to avoid kvdb loss)
+			Step(fmt.Sprintf("set HA level to 2 for app %s's volume", ctx.App.Key), func() {
+				vols, err = Inst().S.GetVolumes(ctx)
+				Expect(err).NotTo(HaveOccurred())
+				setHALevel(vols[0], 2)
+				// ValidateContext() will fail with error "volume has invalid repl value. Expected:3 Actual:2"
+				// without the line below.
+				ctx.SkipVolumeValidation = true
+				replicaNodeIDs = getReplicaNodeIDs(vols[0])
+				Expect(len(replicaNodeIDs)).To(Equal(2))
+			})
+
+			Step(fmt.Sprintf("stop PX on a client node for app %s and wait for export gone", ctx.App.Key), func() {
+				// We need at least 5 nodes to do a failover after stopping PX on the client node.
+				// Otherwise, PX will lose quorum. (This assumes that there are 3 internal kvdb nodes.)
+				failover = len(workers) >= 5
+
+				apiVol, err = Inst().V.InspectVolume(vols[0].ID)
+				Expect(err).NotTo(HaveOccurred())
+
+				attachedNode, err = Inst().V.GetNodeForVolume(vols[0], defaultCommandTimeout, defaultCommandRetry)
+				Expect(err).NotTo(HaveOccurred())
+				logrus.Infof("volume %v (%v) is attached to node %v", vols[0].ID, apiVol.Id, attachedNode.Name)
+
+				// Choose a client node to stop PX on.
+				for _, worker := range workers {
+					// skip replica node since the pod on that node will get terminated.
+					if _, ok := replicaNodeIDs[worker.VolDriverNodeID]; ok {
+						continue
+					}
+					// can't stop 2 metadata nodes
+					if failover && attachedNode.IsMetadataNode && worker.IsMetadataNode {
+						continue
+					}
+					clientNode = &worker
+					break
+				}
+				Expect(clientNode).ToNot(BeNil())
+				logrus.Infof("chose client node %v to stop PX on", clientNode.Name)
+
+				exports := getExportsOnNode(apiVol, attachedNode)
+				Expect(exports).Should(ContainElement(clientNode.DataIp),
+					"client IP not found in the exports before stopping PX on the client node")
+
+				logrus.Infof("stopping volume driver on node %s", clientNode.Name)
+				StopVolDriverAndWait([]node.Node{*clientNode})
+
+				// TODO: offlineClientTimeout = 15 * time.Minute in ref.go
+				// need to make it configurable so that we don't have to sleep for that long in the test
+				logrus.Infof("sleep to allow the server to remove client %v's export", clientNode.Name)
+				time.Sleep(16 * time.Minute)
+
+				// We don't expect a failover
+				validateAttachedNode(vols[0], attachedNode)
+
+				// Verify server removed the export
+				exports = getExportsOnNode(apiVol, attachedNode)
+				Expect(exports).ShouldNot(ContainElement(clientNode.DataIp),
+					"client IP still present in the exports after stopping PX on client node for a long time")
+			})
+
+			Step(fmt.Sprintf("do a failover if possible for app %s", ctx.App.Key), func() {
+				if failover {
+					restartVolumeDriverOnNode(attachedNode)
```

Comment:
Can we do the before and after counter checks? Is that the refactoring you are planning to do?

Reply:
We verify that all the app pods are incrementing their counters after brining up all the nodes. Do you think additional checks are needed?